client/src/providers/UserProvider.tsx

```diff
@@ -2,14 +2,59 @@ import * as React from 'react';
 
 import type { User } from '../types';
 
-type UserContextT = [any | null, (user: any) => void];
+type SetUser = (
+  user: User | null,
+  shouldFetch?: boolean,
+  shouldStartTimer?: boolean,
+) => Promise<void>;
+type UserContextT = [User | null, SetUser];
 
-export const UserContext = React.createContext<UserContextT>([null, (user: any) => {}]);
+export const UserContext = React.createContext<UserContextT>([
+  null,
+  async function (user: User | null, shouldFetch?: boolean, shouldStartTimer?: boolean) {},
```

Comment:
I'm not as versed with React ways as I am with Angular. Why is the function body here empty?

Reply:
yeah it's an odd pattern
this is basically the initial value of the context. we don't ever use it though, what's used is what's passed to the provider below 
`return <UserContext.Provider value={[user, setUserTimeout]}>{children}</UserContext.Provider>;`
so we basically overwrite it on the first render with the `setUserTimeout` timeout function

https://reactjs.org/docs/context.html